### PR TITLE
Update SIG Release teams for 1.21 Release

### PR DESCRIPTION
**Pull request by palnabarun**
- Add RT Lead, EA, RT Lead shadows and Role Leads to
  - `milestone-maintainers`
  - `release-team`
- Add RT Lead, EA, RT Lead Shadows to `release-team-leads`
- Add CI Signal Lead to `ci-signal`

ref: https://github.com/kubernetes/sig-release/issues/1404

/assign
/sig release
/area release-team
/priority critical-urgent
/assign @justaugustus @saschagrunert @jeremyrickard @hasheddan @alejandrox1
/cc @kikisdeliveryservice @savitharaghunathan @bai

### CODE REVIEW

File: config/kubernetes/sig-release/teams.yaml [hunk 1]
```diff
@@ -303,11 +274,11 @@ teams:
               as a notification group. Remove org members who are not current
               Release Team Leads.
             members:
-            - hasheddan # 1.20 RT Lead Shadow
-            - jeremyrickard # 1.20 RT Lead
-            - lachie83 # 1.20 Emeritus Adviser
-            - palnabarun # 1.20 RT Lead Shadow
-            - savitharaghunathan # 1.20 RT Lead Shadow
+            - bai # 1.21 RT Lead Shadow
+            - oikiki # 1.21 Emeritus Adviser
```
Comment (savitharaghunathan) [inline config/kubernetes/sig-release/teams.yaml:278]: Hi Nabarun, Can you replace 1.21 Emeritus Adviser with 1.21 RT Lead Shadow?

File: config/kubernetes/sig-release/teams.yaml [hunk 2]
```diff
@@ -303,11 +274,11 @@ teams:
               as a notification group. Remove org members who are not current
               Release Team Leads.
             members:
-            - hasheddan # 1.20 RT Lead Shadow
-            - jeremyrickard # 1.20 RT Lead
-            - lachie83 # 1.20 Emeritus Adviser
-            - palnabarun # 1.20 RT Lead Shadow
-            - savitharaghunathan # 1.20 RT Lead Shadow
+            - bai # 1.21 RT Lead Shadow
+            - kikisdeliveryservice # 1.21 RT Lead Shadow
```
Comment (jeremyrickard) [inline config/kubernetes/sig-release/teams.yaml:278]: lol you refreshed before i could comment :) :)

File: config/kubernetes/sig-release/teams.yaml (hunk 1 above)
Comment (savitharaghunathan) [reply config/kubernetes/sig-release/teams.yaml]: Also, I think Kirsten's GH id is @kikisdeliveryservice

Comment (palnabarun) [reply config/kubernetes/sig-release/teams.yaml]: Done! 3AM commits are harmful. xD

File: config/kubernetes/sig-release/teams.yaml (hunk 2 above)
Comment (palnabarun) [reply config/kubernetes/sig-release/teams.yaml:278]: 3AM commits are harmful. xD

**Review by hasheddan — APPROVED** [review]
/lgtm
/hold if folks want to verify their comments were addressed

File: config/kubernetes/sig-release/teams.yaml (hunk 1 above)
Comment (kikisdeliveryservice) [reply config/kubernetes/sig-release/teams.yaml]: I got a promotion!! Thank you for your confidence!!!! :wink:

File: config/kubernetes/sig-release/teams.yaml [hunk 3]
```diff
@@ -67,22 +67,19 @@ teams:
     - hpandeycodeit # Usability
     - Huang-Wei # Scheduling
     - idealhack # Release Manager
-    - jameslaverack # 1.20 Release Notes Lead
     - janetkuo # Apps
     - jberkhahn # Service Catalog
     - jberkus # Release
     - jdumars # Architecture / PM
     - jeefy # UI
-    - jeremyrickard # 1.20 RT Lead
```
Comment (justaugustus) [inline config/kubernetes/sig-release/teams.yaml:76]: Leave Jeremy in.

**Review by justaugustus — CHANGES_REQUESTED** [review]
/hold

I might suggest making this PR strictly additive, unless you're positive the contributor in question is:
1. Definitely not returning to the Release Team
2. Will not hold another role that grants them milestone maintainer privileges

With `1`, you may have shadows returning and you'll end having to add them back.
I'd wait until the entire team is locked in to do deletions.

**Comment by palnabarun** [comment]
@justaugustus -- the suggestions makes sense. I've made this changeset strictly additive.

**Comment by justaugustus** [comment]
Here we go!

/lgtm
/approve
/hold cancel

**Pull request merged**